### PR TITLE
perf(Compiler): do not resolve bindings for cached ProtoViews

### LINKED ISSUE #2604: Compiler.compileInHost should first check the cache before creating `DirectiveBinding`
Creating `DirectiveBinding` can be quite expensive and is not needed when the `ProtoView` is cached.

### PR DESCRIPTION
**Pull request by jteplitz**
Fixes #2604

[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2736)

### CODE REVIEW

**Comment by jteplitz602** [comment]
Signed cla

File: modules/angular2/src/core/compiler/compiler.ts [hunk 1]
```diff
@@ -103,16 +104,19 @@ export class Compiler {
   // Create a hostView as if the compiler encountered <hostcmp></hostcmp>.
   // Used for bootstrapping.
   compileInHost(componentTypeOrBinding: Type | Binding): Promise<ProtoViewRef> {
-    var componentBinding = this._bindDirective(componentTypeOrBinding);
-    Compiler._assertTypeIsComponent(componentBinding);
+    var componentType = (isType(componentTypeOrBinding)) ? componentTypeOrBinding
```
Comment (yjbanov) [inline modules/angular2/src/core/compiler/compiler.ts]: No need to wrap `isType` in parens

File: modules/angular2/test/core/compiler/compiler_spec.ts [hunk 2]
```diff
@@ -372,6 +372,25 @@ export function main() {
                async.done();
              });
        }));
+    it('should not bind directives for cached components', inject([AsyncTestCompleter], (async) => {
```
Comment (yjbanov) [inline modules/angular2/test/core/compiler/compiler_spec.ts]: separate tests by 1 blank line

File: modules/angular2/test/core/compiler/compiler_spec.ts [hunk 3]
```diff
@@ -562,6 +581,12 @@ class SpyRenderCompiler extends SpyObject {
   constructor() { super(RenderCompiler); }
   noSuchMethod(m) { return super.noSuchMethod(m) }
 }
+@proxy
```
Comment (yjbanov) [inline modules/angular2/test/core/compiler/compiler_spec.ts]: separate from previous block by 1 blank line

**Comment by yjbanov** [comment]
Make sure travis is happy

**Comment by yjbanov** [comment]
LGTM if LGT Travis

**Pull request closed without merging**